ext/dom/tests/compareDocumentPosition/attribute_child_order.phpt

```diff
@@ -0,0 +1,53 @@
+--TEST--
+compareDocumentPosition: attribute vs child order
+--EXTENSIONS--
+dom
+--FILE--
+<?php
+
+$dom = new DOMDocument();
+$dom->loadXML(<<<XML
+<?xml version="1.0"?>
+<container>
+<before/>
+<outer align="center">
+    <p>foo</p>
+    <div>
+        <div>
+            <p>bar</p>
+        </div>
+    </div>
+</outer>
+</container>
+XML);
+
+$before = $dom->documentElement->firstElementChild;
+$outer = $before->nextElementSibling;
+$foo = $outer->firstElementChild;
+$bar = $foo->nextElementSibling->firstElementChild->firstElementChild;
+
+// See note about attributes vs children positions: attributes precede children
```

Comment:
Not at all confusing :(

Reply:
It kinda makes sense visually. When an element has attributes, we always have to write them down before the children of that element.

Reply:
That's a good point actually